pkg/vault/test/foundry/fuzz/Swap.medusa.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { Rounding } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+
+import { BasePoolMath } from "../../../contracts/BasePoolMath.sol";
+import { BalancerPoolToken } from "../../../contracts/BalancerPoolToken.sol";
+
+import "../utils/BaseMedusaTest.sol";
+
+contract SwapMedusaTest is BaseMedusaTest {
+    uint256 internal constant MIN_SWAP_AMOUNT = 1e6;
+
+    int256 internal invariantDiff = 0;
+
+    constructor() BaseMedusaTest() {
+        vault.manuallySetSwapFee(address(pool), 0);
+    }
+
+    function optimize_currentInvariant() public view returns (int256) {
+        return -int256(invariantDiff);
```

Comment:
You could also optimize the negative of the invariant, instead of the diff. Like:

`return -int256(currentInvariant);`

Where currentInvariant is the invariant of the pool in the end of the swap. This way, the system would search for operations that would reduce the size of the invariant, trying to put it closer to 0.